crux-mir-comp/test/symb_eval/comp/clobber_globals_closure.rs

```diff
@@ -0,0 +1,23 @@
+// A regression test for #2687. rustc is liable to promote the closure used in
+// the body of the foo() function to a static item, which means that the value
+// of this static will be clobbered as part of proving foo_equiv(). Because
+// this closure does not capture any variables, it is represented as an empty
+// MirAggregate in crucible-mir. There was a subtle bug involving mistreatment
+// of empty MirAggregates that #2687 uncovered, and this test case aims to
+// ensure that this bug remains fixed.
+
+extern crate crucible;
+extern crate crucible_proc_macros;
+use crucible::crucible_assert;
+use crucible_proc_macros::crux_spec_for;
+
+pub fn foo() -> u32 {
+    (|| 42u32)()
+}
```

Comment:
I'm a bit wary of having regression tests that rely on promotion.  It seems like it would be really easy to accidentally change the test in a way that makes it no longer test the intended property.  Can you try changing this to ensure that the closure is actually static?  Here are a couple options I can think of for that:
```Rust
// The closure must be promoted to get 'static lifetime
fn mk_closure() -> &'static impl Fn() -> u32 {
    &|| 42u32
}
```
```Rust
#![feature(type_alias_impl_trait)]
type MyClosure = impl Fn() -> u32;
static MY_CLOSURE: MyClosure = || 42u32;
// May need an annotation somewhere to define the hidden type
```

Reply:
Ah, I wasn't aware that you could reliably get `rustc` to emit a static closure like this! I've updated the test to use the `mk_closure()` approach.